annotator/src/main/java/org/cbioportal/annotator/internal/GenomeNexusImpl.java

```diff
@@ -515,19 +521,40 @@ private String resolveEntrezGeneId(boolean replace) {
         if (!replace || canonicalTranscript == null || canonicalTranscript.getGeneId() == null || canonicalTranscript.getGeneId().trim().isEmpty()) {
             return mRecord.getENTREZ_GENE_ID();
         }
+        // check hashmap first before hitting api for gene cross-ref data
+        if (ensemblAccessionEntrezIdMap.containsKey(canonicalTranscript.getGeneId())) {
```

Comment:
I guess it is true that every ensemblAccession maps to only a single EntrezId?

Reply:
yes, as long as it's not a pseudogene or uncharacterized area of the genome then it should be one entrez gene id per ensembl gene accession